pytorch_lightning/trainer/logging.py

```diff
@@ -168,10 +168,6 @@ def process_output(self, output, train=False):
         # ---------------
         hiddens = output.get('hiddens')
 
-        # use every metric passed in as a candidate for callback
-        callback_metrics.update(progress_bar_metrics)
```

Comment:
why do we need to remove this?
without this log metrics and progress bar metrics won't be candidates for the callbacks

Reply:
In #1727 @kessido had the issue, that progress bar or log metric overwrites the callback metric of the top layer dict. An example was also given by @kessido see [COLAB](https://colab.research.google.com/drive/1kYjEH2h5Ki9ygZ0XlA6hpbcxA1L1NNzb?usp=sharing)

I don't know if this needs to be fixed, that's why I asked in the issue for more opinions. Only @awaelchli responded and said he thinks that this also needs to be fixed.

Because no one started a PR I did to initiate a discussion. I have several ideas on how this could be fixed and mentioned some in the issue above. But this was the easiest and quickest solution. I didn't want to spend too much afford on a solution which then will be discarded.

Reply:
@williamFalcon @olineumann, in the current update, when that line is removed and we use Result Obj, we cannot save the model checkpoint in form of  `{val_loss}`, it will result `epoch=1-val_loss=0` which cannot get the `val_loss` due to the filename params based on the `callback_metrics`. Is there another way to assign Result/TrainResul/EvalResult Obj with callback_metrics.